mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java

```diff
@@ -0,0 +1,170 @@
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+
+public class TikaTextExtractor implements TextExtractor {
+
+    private final TikaHttpClient tikaHttpClient;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public TikaTextExtractor(TikaHttpClient tikaHttpClient) {
+        this.tikaHttpClient = tikaHttpClient;
+        this.objectMapper = initializeObjectMapper();
+    }
+
+    private ObjectMapper initializeObjectMapper() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        SimpleModule mapModule = new SimpleModule();
+        mapModule.addDeserializer(ContentAndMetadata.class, new ContentAndMetadataDeserializer());
+        objectMapper.registerModule(mapModule);
+        return objectMapper;
+    }
+
+    @Override
+    public ParsedContent extractContent(InputStream inputStream, String contentType, String fileName) throws Exception {
+        ContentAndMetadata contentAndMetadata = convert(tikaHttpClient.metadataAndContent(inputStream, contentType));
+        return new ParsedContent(contentAndMetadata.getContent(), contentAndMetadata.getMetadata());
+    }
+
+    private ContentAndMetadata convert(String tika) throws IOException, JsonParseException, JsonMappingException {
+        return objectMapper.readValue(tika, ContentAndMetadata.class);
+    }
+
+    private static class ContentAndMetadataDeserializer extends JsonDeserializer<ContentAndMetadata> {
+
+        @Override
+        public ContentAndMetadata deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
+            TreeNode treeNode = jsonParser.getCodec().readTree(jsonParser);
+            Preconditions.checkState(treeNode.isArray() && treeNode.size() == 1, "The response should have only one element");
+            Preconditions.checkState(treeNode.get(0).isObject(), "The element should be a Json object");
+            ObjectNode node = (ObjectNode) treeNode.get(0);
+            return ContentAndMetadata.from(ImmutableList.copyOf(node.fields()).stream()
+                .map(entry -> Pair.of(entry.getKey(), asListOfString(entry.getValue())))
+                .collect(Collectors.toMap(Pair::getLeft, Pair::getRight)));
+        }
+
+        private List<String> asListOfString(JsonNode jsonNode) {
+            if (jsonNode.isArray()) {
+                return ImmutableList.copyOf(jsonNode.elements()).stream()
```

Comment:
But if you collecting you already converting now ? By the way why aren't you directly collecting to immutable list ?